include/openPMD/RecordComponent.hpp

```diff
@@ -119,6 +119,15 @@ class RecordComponent : public BaseRecordComponent
     template< typename T >
     RecordComponent& makeEmpty( uint8_t dimensions );
 
+    /**
+     * @brief Non-template overload of RecordComponent::makeEmpty(). 
+     * Uses the passed openPMD datatype to determine the template parameter.
+     * 
+     * @param dimensions 
+     * @return RecordComponent& 
+     */
+    RecordComponent& makeEmpty( Datatype, uint8_t dimensions );
```

Comment:
I am used to putting that parameter first since that's how `switchType` does it, so I did that without a lot of thinking. You mean because it would align better with the templated `makeEmpty`? I actually like the order that it has now, since you first specify the datatype (as template or as normal parameter), thus also specifying whether you use the templated `makeEmpty` or the non-template, and next you specify the dimensions.
```cpp
rc.makeEmpty< unsigned int >( 3 );
rc.makeEmpty( Datatype::UNSIGNED_INT, 3);
```
Think that's pretty straightforward this way around?

I've completed the doxygen description.

Reply:
Okay, that looks legit as well.
But can we make `makeEmpty` consistent with `makeConstant`?

Reply:
Orthogonal question for a follow-up PR about the return type:
with handles implemented, could we make the coder safer by returning in all those places just a `RecordComponent` instead of a `RecordComponent&`? What do you think?